### PR TITLE
Payments switching two step flow

### PR DESCRIPTION
**Pull request by jonflynng**
## What does this change?

Allows Direct Debit users (currently reviewing UI for paypal users switching into card) to change payment method to card, and also completes the switching in two steps, compared to the current process which takes 3 steps, 4 including the recaptcha screen.

 I've noted that Stripe Elements is buggy with testing libraries and have added an issue on their github.

Currently, this would be merged to a feature branch, and then when the entire feature is completed it would be merged back to main. the feature branch will have to be kept updated with main branch. I'm open to suggestions on a better approach to this. 

## How to test

Click on 'manage payment method' on a subscription in account overview. Use card number '4242 4242 4242 4242' to test successful payments. Stripe has many test numbers to test different error states and also 3D Secure/2 factor authentication: https://stripe.com/docs/testing. For Direct Debit, use account number 55779911 and sort code 200000 to test.

### CODE REVIEW

**Comment by jonflynng** [comment]
[Create two step payment update flow](https://trello.com/c/9hnLM2n2/2694-create-two-step-payment-update-flow)

File: app/client/__tests__/components/payment/cardInputForm.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import React from "react";
+// import { render, fireEvent, waitFor } from "@testing-library/react";
```
Comment (rBangay) [inline app/client/__tests__/components/payment/cardInputForm.tsx:2]: probably don't need this line here ... it should be pretty clear we need to add waitFor back in when/if we get to uncomment out the stripe tests

File: app/client/__tests__/components/payment/currentPaymentDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      const { getByText } = render(
+        <CurrentPaymentDetails {...directDebitProductDetail} />
+      );
+
+      getByText("ending 911");
+      getByText("20-00-00");
+    });
+  });
+
+  describe("For Paypal", () => {});
```
Comment (rBangay) [inline app/client/__tests__/components/payment/currentPaymentDetails.tsx:146]: Do we need this here?

File: app/client/components/OverlayLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	padding: 0;
+	-webkit-tap-highlight-color: transparent;
+	position: fixed;
+	left: 0;
+	top: 0;
+	width: 100%;
+	height: 100%;
+	text-align: center;
+    ${textSans.medium()};
+`
```
Comment (rBangay) [inline app/client/components/OverlayLoader.tsx:27]: It's weird that the pre commit hook didn't pretty this ... maybe `pretty-quick` doesn't know what to do with the template literal string :s

File: app/client/__tests__/components/payment/currentPaymentDetails.tsx (hunk 2 above)
Comment (jonflynng) [reply app/client/__tests__/components/payment/currentPaymentDetails.tsx:146]: I filled this in in the third PR

**Review by rBangay — APPROVED** [review]

File: app/client/components/OverlayLoader.tsx (hunk 3 above)
Comment (rBangay) [reply app/client/components/OverlayLoader.tsx:27]: as an update to this. We needed to update the version of pretty-quick after bumping our node version

**Review by rBangay — APPROVED** [review]

**Pull request merged**